src/functional/ffront/func_to_foast.py

```diff
@@ -362,14 +362,14 @@ def visit_ImportFrom(self, node: ast.ImportFrom, **kwargs) -> None:
 
         return foast.ExternalImport(symbols=symbols, location=self._make_loc(node))
 
-    def visit_arguments(self, node: ast.arguments) -> list[foast.FieldSymbol]:
+    def visit_arguments(self, node: ast.arguments) -> list[foast.Symbol]:
         return [self.visit_arg(arg) for arg in node.args]
 
-    def visit_arg(self, node: ast.arg) -> foast.FieldSymbol:
+    def visit_arg(self, node: ast.arg) -> foast.Symbol:
```

Comment:
Same comment here. Also, shouldn't it be checked that `new_type` is a `DataType`?

Reply:
If `new_type` is not a DataType pydantic will already throw an error. If added an additional check that gives a more readable message.